document/js-api/index.bs

```diff
@@ -1329,6 +1383,165 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
+        1. Return the result of performing [=coerce a JavaScript return=] on |results| and |ret|.
+    1. Let |store| be the [=surrounding agent=]'s [=associated store=].
+    1. Let (|store|, |funcaddr|) be [=func_alloc=](|store|, |functype|, |hostfunc|).
+    1. Set the [=surrounding agent=]'s [=associated store=] to |store|.
+    1. Return |funcaddr|.
+</div>
+
+<div algorithm>
+The <dfn>Pause</dfn> abstract operation takes a currently executing algorithm and pauses it. The most important parameter |ec| of this operation is an [=Execution Context=] which must be the most recent entry in the [=Execution Context Stack=] and which defines which operation is being paused. 
+
+The result of pausing an operation is a continuation |k| and a newly surfaced [=Execution Context=] -- which must be the [=Execution Context=] immediately beneath the one on the [=Execution Context Stack=] that is [=paused=].
+
+    1. If |ec| is not the most recent entry on the [=Execution Context Stack=] then trap.
+    1. Let |map| be the [=surrounding agent=]'s associated [=Execution Context Status map=].
+    1. If the entry for |ec| in |map| is not [=active=] then trap.
+    1. Let |k| be a continuation that represents the remainder of the computation associated with |ec|.
+    1. Remove |ec| from the [=Execution Context Stack=].
+    1. Set the entry for |ec| in |map| to [=paused=].
+    1. Assert that the [=Execution Context Stack=] is not empty.
+    1. Return |k|
+</div>
+
+<div algorithm>
+The <dfn>Enter</dfn> abstract operation takes a continuation object |k|, a value object |v| and an [=execution context=] |ec|, and resumes the operations defined by |k|.
+    1. Let |map| be the [=surrounding agent=]'s associated [=Execution Context Status map=].
+    1. If the entry for |ec| in |map| is not [=paused=] then trap.
+    1. Set the entry for |ec| in |map| to [=active=].
+    1. Push |ec| on to the [=surrounding agent=]'s [=Execution Context Stack=], making |ec| the current execution context.
+    1. Resume the operations defined by |k|, passing |v| as the value of the last instruction being performed by |k|
+    1. Return undefined
+</div>
+
+<div algorithm>
+The <dfn>Reject</dfn> abstract operation takes a continuation object |k|, an exception object |e| and an [=execution context=] |ec|, and resumes the operations defined by |k| by throwing an exception to the [=paused=] computation.
+    1. Let |map| be the [=surrounding agent=]'s associated [=Execution Context Status map=].
+    1. If the entry for |ec| in |map| is not [=paused=] then trap.
+    1. Set the entry for |ec| in |map| to [=active=].
+    1. Push |ec| on to the [=surrounding agent=]'s [=Execution Context Stack=], making |ec| the current execution context.
+    1. Let |type|, |payload| and |opaqueData| be the result of performing [=coerce a JavaScript exception=] on |e|
+    1. Resume the operations defined by |k|, performing [=WebAssembly/Throw=] with |type|, |payload| and |opaqueData|.
```

Comment:
What does it mean to resume |k| but also throw?

Reply:
Resuming with a throw. In the typed continuations proposal there is (will be) a specific operation that is equivalent; also called resume_throw.